packages/engine/lib/stateful/src/field/mod.rs

```diff
@@ -1,4 +1,11 @@
-//! TODO: DOC
+//! Provides an hEngine-specific representation of Arrows `Field` and `DataType`.
+//!
+//! As we are using only a subset of types provided by `Arrow`, our types are represented by
+//! [`FieldType`] rather than [`arrow::DataType`]. As [`FieldType`] can't be used in
+//! [`arrow::Field`], this module provides [`FieldSpec`] instead.
+//!
+//! For naming a [`FieldSpec`], [`FieldKey`] is provided, as a name has strict requirements on how
+//! it's built.
```

Comment:
I would go even further with this to say that `FieldKey`s provide unique identifiers for fields, and encapsulate information to do with the scope of data-access and the source of the component that manages it.

Reply:
Added that